### PR TITLE
Layout01

### PR DESCRIPTION
**Pull request by Sushmareddy-OSI**
Added a component `Layout01`

### CODE REVIEW

File: src/test/Layout01_test.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    const renderedDOM = () => React.findDOMNode(layout1);
+    
+    it('renders the tag', function() {
+        expect(result_tag.type).toBe('div')
+    })
+
+    it("find element with tag", function() {
+
+        var root = TestUtils.renderIntoDocument(<Layout1/>);
+        // scryRenderedComponentsWithType
```
Comment (edgibbs) [inline src/test/Layout01_test.js]: comment should be removed

File: README.md [hunk 2]
```diff
@@ -27,12 +27,13 @@ The react-wood-duck is a shared pattern library hosting re-usable components and
...
 *	GlobalHeader
 *	InputComponent
-*	ListItem
+*	AddAdditionalItem
 *	Select/Multiselect
 *       SideNav
 *	TextArea
 *	Table
 *   PageHeader
+*	Layout01
```
Comment (edgibbs) [inline README.md]: Is Layout01 the name from DesignOps?

Comment (Sushmareddy-OSI) [reply README.md]: I don't know... In story it is mentioned as `01 landing page template` so I made it as layout01.I am not sure about the specific name.

Comment (NagaPradeepGajula-osi) [reply README.md]: I spoke with @ghill-osi about the name of component for 01 landing page and he said that he is ok with layout01.

Comment (ghill-osi) [reply README.md]: ok by me. open to other names, but it works for now.

File: src/test/Layout01_test.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        var compglbHeader = TestUtils.scryRenderedComponentsWithType(root, GlobalHeader);
+        expect(compglbHeader.length).toEqual(1);
+        var pgeHeader = TestUtils.scryRenderedComponentsWithType(root, PageHeader);
```
Comment (edgibbs) [inline src/test/Layout01_test.js]: probably just pageHeader versus pgeHeader.

File: src/test/Layout01_test.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import Layout1 from '../Layout01.js';
+import GlobalHeader from '../GlobalHeader.js';
+import PageHeader from '../PageHeader.js';
+import TestUtils from 'react-dom/lib/ReactTestUtils';
+
+describe('Layout1', function() {
```
Comment (edgibbs) [inline src/test/Layout01_test.js]: This was my refactoring:

``` javascript
import React from 'react';
import Layout1 from '../Layout01.js';
import GlobalHeader from '../GlobalHeader.js';
import PageHeader from '../PageHeader.js';
import TestUtils from 'react-dom/lib/ReactTestUtils';

describe('Layout1', function() {
  it('renders the tag', function() {
    const renderer = TestUtils.createRenderer();
    renderer.render(<Layout1 />);
    const resultTag = renderer.getRenderOutput();
    expect(resultTag.type).toBe('div');
  });

  it('includes GlobalHeader and PageHeader', function() {
    const layout = TestUtils.renderIntoDocument(<Layout1 />);
    expect(
      TestUtils.findRenderedComponentWithType(layout, GlobalHeader)
    ).toBeTruthy();
    expect(
      TestUtils.findRenderedComponentWithType(layout, PageHeader)
    ).toBeTruthy();
  });
});

```

File: src/test/Layout01_test.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    const renderedComp = TestUtils.createRenderer();
+    const compRendered = renderedComp.render(<Layout1 />);
```
Comment (edgibbs) [inline src/test/Layout01_test.js]: The return value here isn't needed.

File: src/test/Layout01_test.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    const resultTag = renderedComp.getRenderOutput();
+    const layout = TestUtils.renderIntoDocument(<Layout01 />);
+    
+    it('renders the tag', function() {
+        expect(resultTag.type).toBe('div')
+    })
+
+    it("find if it contains GlobalHeader and PageHeader", function() {
+        const compglbHeader = TestUtils.scryRenderedComponentsWithType(layout, GlobalHeader);
+        expect(compglbHeader.length).toEqual(1);
```
Comment (edgibbs) [inline src/test/Layout01_test.js]: So this test is functionally testing that the Global Header is contained in the rendered document, but I found it was a little clearer with this style:

```javascript
    expect(                                                                                                                                                 |@
      TestUtils.scryRenderedComponentsWithType(layout, GlobalHeader)                                                                                        |@
    ).toBeTruthy(); 
```
or this style:

```javascript
    const globalHeader = TestUtils.scryRenderedComponentsWithType(                                                                                    
      layout,                                                                                                                                               
      GlobalHeader                                                                                                                                          
    )[0];
    expect(                                                                                                                                                 
      TestUtils.isCompositeComponentWithType(globalHeader, GlobalHeader)                                                                                
    ).toEqual(true);  
```

versus testing that the length of the collection received back has a length of 1:

```javascript
    const compglbHeader = TestUtils.scryRenderedComponentsWithType(layout, GlobalHeader);
    expect(compglbHeader.length).toEqual(1);
```

File: src/test/Layout01_test.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        const pgeHeader = TestUtils.scryRenderedComponentsWithType(layout, PageHeader);
```
[the remark this answers is not in the document]
Comment (ghill-osi) [reply src/test/Layout01_test.js]: i concur. unless it takes a sentence to describe something, spell it out completely to rule out any confusion.

File: src/Layout01.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import GlobalHeader from './GlobalHeader.js';
+import PageHeader from './PageHeader.js';
+
+class Layout01 extends React.Component {
+  constructor(props) {
```
Comment (ratneshraval) [inline src/Layout01.js]: unnecessary constructor.

Comment (Sushmareddy-OSI) [reply src/Layout01.js]: We just placed it as a placeholder for future iterations as we might need it when we pass the props to the main content area.

**Comment by edgibbs** [comment]
Needs quick storybook story.

**Pull request closed without merging**